test/jdk/javax/swing/JFileChooser/bug4943900.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2004, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4943900
+ * @summary Tests that FileFilter combo box is shown in FileChooser
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jtreg.SkippedException
+ * @run main/manual bug4943900
+ */
+
+import java.io.File;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.UIManager;
+import javax.swing.filechooser.FileFilter;
+import jtreg.SkippedException;
+
+public class bug4943900 {
+    private static final String INSTRUCTIONS = """
+        1. Click "Show FileChooser" button below, FileChooser dialog must
+           appear.
```

Comment:
I do not see the "Show FileChooser" button, the file chooser showed automatically.

However, it is shown in an unobvious way (code-wise), via the `frame.add(fc);` call (and later the `setVisible(true)` call on the JFrame shows it).

Is this intentional?

Reply:
Updated instructions.
I previously had "Show FileChooser" button on click of which jfc was shown but during the course of review got a suggestion to remove the extra button click and directly add it to the frame so that it doesn't cover the instruction UI.

Reply:
Updated to remove redundant frame and simply return jfc.